libbeat/outputs/kafka/config.go

```diff
@@ -112,7 +112,8 @@ var compressionModes = map[string]sarama.CompressionCodec{
 // validTopicRegExp is used to validate the topic contains only valid characters
 // when running under Elastic-Agent. The regexp is taken from:
 // https://github.com/apache/kafka/blob/a126e3a622f2b7142f3543b9dbee54b6412ba9d8/clients/src/main/java/org/apache/kafka/common/internals/Topic.java#L33
-var validTopicRegExp = regexp.MustCompile("^[a-zA-Z0-9._-]+$")
+// also allowing a format string that contains a field name %{[FIELD]}
+var validTopicRegExp = regexp.MustCompile(`^(?:%\{\[)?[a-zA-Z0-9._-]+(?:\]\})?$`)
```

Comment:
I think we can probably get rid of this, it was meant to catch people including the topic selector syntax before it hit Kafka and matched the valid kafka topic syntax exactly.

If we were to keep the valid topic regex, the place to put it would be after the topic substitution has already happened.

Reply:
removed the validation

Reply:
We can probably just remove it entirely, this validation also exists in the Kafka API when we go to create the topic.

We were trying to catch the problem earlier with this and make things more obvious if users pasted examples that used to work when we disabled it.

If we did keep it we'd put in the block below, which is at the point where he message is published and so is per event and doesn't buy us anything over what the Kafka API is going to give us.

https://github.com/elastic/beats/blob/d7ae68c7aa40aa27deff6b90508bb8f8664a6efa/libbeat/outputs/kafka/client.go#L215-L221

Reply:
Thanks, I'll leave it removed then.